sqlglot/schema.py

```diff
@@ -451,16 +447,17 @@ def normalize_name(
     dialect: DialectType = None,
     is_table: bool = False,
     normalize: t.Optional[bool] = True,
-) -> str:
+) -> exp.Identifier:
     if isinstance(identifier, str):
         identifier = exp.parse_identifier(identifier, dialect=dialect)
 
-    if not normalize:
-        return identifier.name
-
     # This can be useful for normalize_identifier
     identifier.meta["is_table"] = is_table
```

Comment:
I think this needs to be moved after the following branch?

Reply:
why?

Reply:
Because IIRC it's only needed for normalizations, right? If we won't normalize then why should we mutate here?